### PR TITLE
Pin nightly to a specific date

### PR DESCRIPTION
**Pull request by jack-signal**
I picked 11/1 simply because that was what has been used for Java builds so far.

### CODE REVIEW

**Comment by jack-signal** [comment]
Oh fmt and clippy aren't available for this nightly, probably need those... will pick another date

**Comment by jack-signal** [comment]
Per https://rust-lang.github.io/rustup-components-history/index.html 2020-11-09 is the only day this week that has had all components available.

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly
+nightly-2020-11-01
```
Comment (jrose-signal) [inline rust-toolchain]: I don't think the GitHub Action is respecting this, but it shouldn't be too hard to read from this file to do that (cf. me [doing the same for Node](https://github.com/signalapp/libsignal-client/pull/20/files#diff-4e88a8cc63d2540befa3412886ac9b9db5cd10bce852960298517ea4bd7f7bb7R26-R31)).

File: .github/workflows/java.yml [hunk 2]
```diff
@@ -19,7 +19,6 @@ jobs:
     - name: Install nightly rust
       uses: actions-rs/toolchain@v1
       with:
-        toolchain: nightly
```
Comment (jrose-signal) [inline .github/workflows/java.yml:22]: Oh, that's great that that just works! I didn't even check.

**Review by jrose-signal — APPROVED** [review]

File: .github/workflows/java.yml (hunk 2 above)
Comment (jack-signal) [reply .github/workflows/java.yml:22]: It's documented as working for the action, but isn't finding the `rust-toolchain` file for some reason, further debugging required.

Comment (jrose-signal) [reply .github/workflows/java.yml:22]: Looks like we've just got the steps in the wrong order—checkout first, then install. We haven't had to do that before now.

Comment (jack-signal) [reply .github/workflows/java.yml:22]: I wasn't able to get it to read a rust-toolchain even with a simple repo so I opened an issue with actions-rs https://github.com/actions-rs/toolchain/issues/123. I'm going to wait to hear back from them, or if nothing then I'll try the approach you used for node.

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -16,13 +16,13 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Install nightly rust
+    - uses: actions/checkout@v2
+    - name: Install Rust
       uses: actions-rs/toolchain@v1
       with:
         override: true
```
Comment (jrose-signal) [inline .github/workflows/rust.yml:23]: Nitpick: Missed `override: true` (though it shouldn't matter)

**Pull request merged**